src/Sulu/Bundle/RouteBundle/Generator/RouteGeneratorPool.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\Generator;
+
+use Sulu\Bundle\RouteBundle\Exception\MissingClassMappingConfigurationException;
+use Sulu\Bundle\RouteBundle\Model\RoutableInterface;
+
+/**
+ * Using route-generators and configuration to generate route.
+ */
+class RouteGeneratorPool implements RouteGeneratorPoolInterface
+{
+    /**
+     * @var array
+     */
+    private $mappings;
+
+    /**
+     * @var RouteGeneratorInterface[]
+     */
+    private $routeGenerators;
+
+    /**
+     * @param array $mappings
+     * @param RouteGeneratorInterface[] $routeGenerators
+     */
+    public function __construct(array $mappings, array $routeGenerators)
+    {
+        $this->mappings = $mappings;
+        $this->routeGenerators = $routeGenerators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate(RoutableInterface $entity, $path = null)
```

Comment:
Would probably still be better than Pool... But why can't you also use `$options` instead of `$path`? Although I don't like the idea of options, it would have the advantage that the interface is the same, and the additional interface would not be necessary. Also, the caller wouldn't have to care if he's talking to a `ChainRouteGenerator` or a simple one.